Cron/KlSyncs.php

```diff
@@ -124,29 +126,37 @@ private function sendUpdatesToApp(array $groupedRows, bool $isRetry = false)
 
             if (in_array($topic, $trackApiTopics) && !empty($rows)) {
                 foreach($rows as $row) {
-                    $decodedPayload = json_decode($row['payload'], true);
-
-                    $eventTime = $decodedPayload['time'];
-                    unset($decodedPayload['time']);
-
-                    //TODO: if conditional for backward compatibility, needs to be removed in future versions
-                    $storeId = '';
-                    if (isset($decodedPayload['StoreId']))
-                    {
-                        $storeId = $decodedPayload['StoreId'];
-                        unset($decodedPayload['StoreId']);
+                    try {
+                        $decodedPayload = json_decode($row['payload'], true);
```

Comment:
Can we check the return here? It shoudl be false or null if truncated and we cannot decode. Can log specifically for that.

Reply:
If we know we've got a bad payload can we skip sending to Klaviyo in the first place?